game-core/src/test/java/games/strategy/triplea/delegate/MatchesTest.java

```diff
@@ -254,4 +267,180 @@ void shouldNotMatchWhenSeaTerritoryIsUnownedAndDoesNotHaveTerritoryAttachment()
+          "An infrastructure unit with attack can be in battle when it is attacking",
+          Matches.unitCanBeInBattle(true, true, 1, false, List.of()).test(unit),
+          is(true));
+    }
+
+    @Test
+    void infrastructureWithAttackCanNotBeInBattleWhenDefending() {
+      when(gameData.getDiceSides()).thenReturn(6);
+      final UnitType unitType = new UnitType("infrastructure", gameData);
+      final UnitAttachment unitAttachment =
+          new UnitAttachment("infrastructure", unitType, gameData);
+      unitAttachment.setIsInfrastructure(true);
+      unitAttachment.setAttack(1);
+      unitType.addAttachment(UNIT_ATTACHMENT_NAME, unitAttachment);
+      final Unit unit = unitType.create(1, player, true).get(0);
+
+      assertThat(
+          "An infrastructure unit with attack can not be in battle when it is attacking",
+          Matches.unitCanBeInBattle(false, true, 1, false, List.of()).test(unit),
+          is(false));
+    }
+
+    @Test
+    void infrastructureWithDefenseCanBeInBattleWhenDefending() {
+      when(gameData.getDiceSides()).thenReturn(6);
+      final UnitType unitType = new UnitType("infrastructure", gameData);
+      final UnitAttachment unitAttachment =
+          new UnitAttachment("infrastructure", unitType, gameData);
+      unitAttachment.setIsInfrastructure(true);
+      unitAttachment.setDefense(1);
+      unitType.addAttachment(UNIT_ATTACHMENT_NAME, unitAttachment);
+      final Unit unit = unitType.create(1, player, true).get(0);
+
+      assertThat(
+          "An infrastructure unit with defense can be in battle when it is defending",
+          Matches.unitCanBeInBattle(false, true, 1, false, List.of()).test(unit),
+          is(true));
+    }
+
+    @Test
+    void infrastructureWithDefenseCanNotBeInBattleWhenAttacking() {
+      when(gameData.getDiceSides()).thenReturn(6);
+      final UnitType unitType = new UnitType("infrastructure", gameData);
+      final UnitAttachment unitAttachment =
+          new UnitAttachment("infrastructure", unitType, gameData);
+      unitAttachment.setIsInfrastructure(true);
+      unitAttachment.setDefense(1);
+      unitType.addAttachment(UNIT_ATTACHMENT_NAME, unitAttachment);
+      final Unit unit = unitType.create(1, player, true).get(0);
+
+      assertThat(
+          "An infrastructure unit with defense can not be in battle when it is attacking",
+          Matches.unitCanBeInBattle(true, true, 1, false, List.of()).test(unit),
+          is(false));
+    }
+
+    @Test
+    void infrastructureThatGivesAnyTypeOfSupportCanBeInBattle() {
+      final UnitType unitType = new UnitType("infrastructure", gameData);
+      final UnitAttachment unitAttachment =
+          new UnitAttachment("infrastructure", unitType, gameData);
+      unitAttachment.setIsInfrastructure(true);
+      final UnitSupportAttachment unitSupportAttachment =
+          new UnitSupportAttachment(
+              Constants.SUPPORT_ATTACHMENT_PREFIX + "support", unitType, gameData);
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Constants.SUPPORT_ATTACHMENT_PREFIX' due to existing static import 'games.strategy.triplea.Constants.SUPPORT_ATTACHMENT_PREFIX'](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6335347)